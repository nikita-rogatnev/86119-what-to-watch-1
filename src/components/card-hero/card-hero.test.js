import React from 'react';
import renderer from 'react-test-renderer';

import CardHero from './card-hero';

import mockData from '../../mocks/mock-data';

jest.mock(`../header/header`, () => `Header`);

it(`CardHuge renders correctly`, () => {
  const tree = renderer
    .create(<CardHero
      data={mockData}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
