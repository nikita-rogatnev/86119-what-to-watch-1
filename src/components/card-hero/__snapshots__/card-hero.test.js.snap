// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHero renders correctly 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__hero"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Matrix"
        src="https://es31-server.appspot.com/wtw/static/film/background/matrix.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <ConnectFunction />
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          Matrix
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Action
          </span>
          <span
            className="movie-card__year"
          >
            1999
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <button
            className="btn btn--play movie-card__button"
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </button>
          <button
            className="btn btn--list movie-card__button"
            type="button"
          >
            <svg
              height="14"
              viewBox="0 0 18 14"
              width="18"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
          <Link
            className="btn movie-card__button"
            to="undefined/review"
          >
            Add review
          </Link>
        </div>
      </div>
    </div>
  </div>
  <div
    className="movie-card__wrap movie-card__translate-top"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster movie-card__poster--big"
      >
        <img
          alt="Matrix poster"
          height="327"
          src="https://es31-server.appspot.com/wtw/static/film/poster/matrix.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <nav
          className="movie-nav movie-card__nav"
        >
          <ul
            className="movie-nav__list"
          >
            <li
              className="movie-nav__item"
            >
              <NavLink
                className="movie-nav__link"
                exact={true}
                to="/films/1/"
              >
                Overview
              </NavLink>
            </li>
            <li
              className="movie-nav__item"
            >
              <NavLink
                className="movie-nav__link"
                to="/films/1/details"
              >
                Details
              </NavLink>
            </li>
            <li
              className="movie-nav__item"
            >
              <NavLink
                className="movie-nav__link"
                to="/films/1/reviews"
              >
                Reviews
              </NavLink>
            </li>
          </ul>
        </nav>
        <Switch>
          <Route
            exact={true}
            path="/films/1/"
            render={[Function]}
          />
          <Route
            path="/films/1/details"
            render={[Function]}
          />
          <Route
            path="/films/1/reviews"
            render={[Function]}
          />
        </Switch>
      </div>
    </div>
  </div>
</section>
`;
